app/Models/Collection.php

```diff
@@ -417,4 +417,22 @@ public function recentlyViewed(): bool
 
         return $this->last_viewed_at->gte(now()->subDays(1));
     }
+
+    /**
+     * A collection is only *potentially* full if the total supply matches the last token number that was indexed for the collection.
+     *
+     * Or if the number of NFTs in a collection matches the total supply.
+     */
+    public function isPotentiallyFull(): bool
+    {
+        if ($this->last_indexed_token_number === null || $this->supply === null) {
+            return false;
+        }
+
+        if ((string) $this->supply === (string) $this->last_indexed_token_number) {
+            return true;
+        }
+
+        return $this->supply === $this->nfts()->count();
```

Comment:
this is also an interesting one as the count can be more than the supply as well, meaning that we indexed more than the contract indicates there should be (it doesn't happen that often, but it's for example the case for `Cool Cats Fractures` where the database count is 11k and the supply is 5986

Reply:
logic wise maybe the following:
- if supply === last token number => full (handles general case where 10k supply results in 10k token numbers)
- if supply === last token number - 1 && db count === supply => full (handles cases where token number starts from 0)
- if supply < last token number && db count === supply => full (handles cases where token numbers are arbitrary values, usually much larger values than the supply is)
- if supply > db count => not full (for cases where we have supply < count (e.g. burn metrics)). The assumption we will make here is that we can handle the burn scenario by returning `full` by default if this last check does not match, as they generally start with their full collection at first (e.g. cool cats fractures at 11.5k) and burn over time, causing the supply to decrease over time while we already indexed them all

Reply:
@crnkovic ^